&max-size unusable with const

Example `test.spicy`:

```spicy
module Test;

const MaxStringSize = 13;

public type NullTerminatedString = unit {
    s: bytes &until=b"\x00" &max-size=MaxStringSize;
};
```

Fails like this for me:

```
$ printf 'testing\000' | spicy-dump test.spicy
/tmp/hilti-rt-test-VLFCHX/Test.cc: In member function ‘std::tuple<hilti::rt::stream::View, SafeInt<long int, hilti::rt::integer::detail::SafeIntException>, hilti::rt::stream::SafeConstIterator> __hlt::Test::NullTerminatedString::__parse_Test_NullTerminatedString_stage2(hilti::rt::ValueReference<hilti::rt::Stream>&, hilti::rt::stream::View, hilti::rt::Bool, hilti::rt::integer::safe<long int>, hilti::rt::stream::SafeConstIterator)’:
/tmp/hilti-rt-test-VLFCHX/Test.cc:144:68: error: passing ‘hilti::rt::integer::safe<long unsigned int>’ {aka ‘const SafeInt<long unsigned int, hilti::rt::integer::detail::SafeIntException>’} as ‘this’ argument discards qualifiers [-fpermissive]
  144 |     hilti::rt::stream::View __limited = __cur.limit(++__hlt::Test::MaxStringSize);
      |                                                                    ^~~~~~~~~~~~~
In file included from /usr/local/spicy/include/hilti/rt/safe-int.h:6,
                 from /usr/local/spicy/include/hilti/rt/types/interval.h:14,
                 from /usr/local/spicy/include/hilti/rt/types/time.h:12,
                 from /usr/local/spicy/include/hilti/rt/util.h:23,
                 from /usr/local/spicy/include/hilti/rt/configuration.h:10,
                 from /usr/local/spicy/include/hilti/rt/libhilti.h:10,
                 from /tmp/hilti-rt-test-VLFCHX/Test.cc:6:
/usr/local/spicy/include/hilti/rt/3rdparty/SafeInt/SafeInt.hpp:5901:35: note:   in call to ‘constexpr SafeInt<T, E>& SafeInt<T, E>::operator++() [with T = longunsigned int; E = hilti::rt::integer::detail::SafeIntException]’
 5901 |     _CONSTEXPR14 SafeInt< T, E >& operator ++() SAFEINT_CPP_THROW
      |                                   ^~~~~~~~
/tmp/hilti-rt-test-VLFCHX/Test.cc:145:67: error: passing ‘hilti::rt::integer::safe<long unsigned int>’ {aka ‘const SafeInt<long unsigned int, hilti::rt::integer::detail::SafeIntException>’} as ‘this’ argument discards qualifiers [-fpermissive]
  145 |     hilti::rt::stream::View __ncur = __cur.advance(++__hlt::Test::MaxStringSize);
      |                                                                   ^~~~~~~~~~~~~
In file included from /usr/local/spicy/include/hilti/rt/safe-int.h:6,
                 from /usr/local/spicy/include/hilti/rt/types/interval.h:14,
                 from /usr/local/spicy/include/hilti/rt/types/time.h:12,
                 from /usr/local/spicy/include/hilti/rt/util.h:23,
                 from /usr/local/spicy/include/hilti/rt/configuration.h:10,
                 from /usr/local/spicy/include/hilti/rt/libhilti.h:10,
                 from /tmp/hilti-rt-test-VLFCHX/Test.cc:6:
/usr/local/spicy/include/hilti/rt/3rdparty/SafeInt/SafeInt.hpp:5901:35: note:   in call to ‘constexpr SafeInt<T, E>& SafeInt<T, E>::operator++() [with T = longunsigned int; E = hilti::rt::integer::detail::SafeIntException]’
 5901 |     _CONSTEXPR14 SafeInt< T, E >& operator ++() SAFEINT_CPP_THROW
      |                                   ^~~~~~~~
[error] spicy-dump: aborting after errors
```

--- a/spicy/toolchain/src/compiler/codegen/parser-builder.cc
+++ b/spicy/toolchain/src/compiler/codegen/parser-builder.cc
@@ -600,7 +600,10 @@ struct ProductionVisitor
             length = builder::coerceTo(*a->valueAs<Expression>(), type::UnsignedInteger(64));
         if ( auto a = AttributeSet::find(field->attributes(), "&max-size") )
             // Append a sentinel byte for `&max-size` so we can detect reads beyond the expected length.
-            length = builder::incrementPrefix(builder::coerceTo(*a->valueAs<Expression>(), type::UnsignedInteger(64)));
+            length =
+                builder()->addTmp("max_size",
+                                  builder::sum(builder::coerceTo(*a->valueAs<Expression>(), type::UnsignedInteger(64)),
+                                               builder::integer(1)));
 
         if ( length ) {
             // Limit input to the specified length.
@@ -955,7 +958,10 @@ struct ProductionVisitor
             length = builder::coerceTo(*a->valueAs<Expression>(), type::UnsignedInteger(64));
         else if ( auto a = AttributeSet::find(p.unitType().attributes(), "&max-size") )
             // Append a sentinel byte for `&max-size` so we can detect reads beyond the expected length.
-            length = builder::incrementPrefix(builder::coerceTo(*a->valueAs<Expression>(), type::UnsignedInteger(64)));
+            length =
+                builder()->addTmp("max_size",
+                                  builder::sum(builder::coerceTo(*a->valueAs<Expression>(), type::UnsignedInteger(64)),
+                                               builder::integer(1)));
 
         if ( length ) {
             // Limit input to the specified length.

--- a/tests/Baseline/spicy.types.unit.field-max-size/output
+++ b/tests/Baseline/spicy.types.unit.field-max-size/output
@@ -2,6 +2,6 @@
 done, [$xs=b"\x00"]
 done, [$xs=b"\x01\x00"]
 error, [$xs=(not set)]
-[error] terminating with uncaught exception of type spicy::rt::ParseError: parse error: parsing not done within &max-size bytes (<...>/field-max-size.spicy:13:40)
+[error] terminating with uncaught exception of type spicy::rt::ParseError: parse error: parsing not done within &max-size bytes (<...>/field-max-size.spicy:15:40)
 error, [$xs=b"\x01\x01\x01\x00"]
-[error] terminating with uncaught exception of type spicy::rt::ParseError: parse error: end-of-data reached before &until-including expression found (<...>/field-max-size.spicy:13:9)
+[error] terminating with uncaught exception of type spicy::rt::ParseError: parse error: end-of-data reached before &until-including expression found (<...>/field-max-size.spicy:15:9)

--- a/tests/Baseline/spicy.types.unit.unit-max-size/output
+++ b/tests/Baseline/spicy.types.unit.unit-max-size/output
@@ -3,6 +3,6 @@ done, [$xs=b"\x00"]
 done, [$xs=b"\x01\x00"]
 done, [$xs=b"\x01\x01\x00"]
 error, [$xs=b"\x01\x01\x00"]
-[error] terminating with uncaught exception of type spicy::rt::ParseError: parse error: parsing not done within &max-size bytes (<...>/unit-max-size.spicy:17:3)
+[error] terminating with uncaught exception of type spicy::rt::ParseError: parse error: parsing not done within &max-size bytes (<...>/unit-max-size.spicy:19:3)
 error, [$xs=b"\x01\x01\x01\x00"]
-[error] terminating with uncaught exception of type spicy::rt::ParseError: parse error: end-of-data reached before &until-including expression found (<...>/unit-max-size.spicy:13:9)
+[error] terminating with uncaught exception of type spicy::rt::ParseError: parse error: end-of-data reached before &until-including expression found (<...>/unit-max-size.spicy:15:9)

--- a/tests/spicy/types/unit/field-max-size.spicy
+++ b/tests/spicy/types/unit/field-max-size.spicy
@@ -9,8 +9,10 @@
 
 module Mini;
 
+const MaxSize = 2;
+
 public type Test = unit {
-    xs: bytes &until-including=b"\x00" &max-size=2;
+    xs: bytes &until-including=b"\x00" &max-size=MaxSize;
 
     on %done { print "done", self; }
     on %error { print "error", self; }

--- a/tests/spicy/types/unit/unit-max-size.spicy
+++ b/tests/spicy/types/unit/unit-max-size.spicy
@@ -9,9 +9,11 @@
 
 module Mini;
 
+const MaxSize = 2;
+
 public type Foo = unit {
     xs: bytes &until-including=b"\x00";
 
     on %done { print "done", self; }
     on %error { print "error", self; }
-} &max-size=2;
+} &max-size=MaxSize;

